pkg/runtime.v2/core/trainingruntime.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2024 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	jobsetv1alpha2 "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+
+	kubeflowv2 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1"
+	runtime "github.com/kubeflow/training-operator/pkg/runtime.v2"
+	fwkcore "github.com/kubeflow/training-operator/pkg/runtime.v2/framework/core"
+	fwkplugins "github.com/kubeflow/training-operator/pkg/runtime.v2/framework/plugins"
+	idxer "github.com/kubeflow/training-operator/pkg/runtime.v2/indexer"
+)
+
+var (
+	errorNotFoundSpecifiedTrainingRuntime = errors.New("not found TrainingRuntime specified in TrainJob")
```

Comment:
```suggestion
	errorNotFoundSpecifiedTrainingRuntime = errors.New("TrainingRuntime is not specified in TrainJob")
```

Reply:
Actually, this message indicates to the "TrainingRuntime is not found". So I rephrased to the "TrainingRuntime specified in TrainJob is not found"

Reply:
Done.